src/PubSub/BatchPublishContainer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\PubSub;
+
+/**
+ * A container whose job is to execute queued items from the batch.
+ *
+ * @internal
+ */
+class BatchPublishContainer
+{
+    /**
+     * @var array
+     */
+    private static $topics = [];
+
+    /**
+     * @var string
+     */
+    private $topicName;
+
+    /**
+     * @var array
+     */
+    private $clientConfig;
+
+    /**
+     * @var bool
+     */
+    private $debugOutput;
+
+    /**
+     * @param string $topicName
+     * @param array $clientConfig
+     * @param bool $debugOutput
+     */
+    public function __construct($topicName, array $clientConfig, $debugOutput)
+    {
+        $this->topicName = $topicName;
+        $this->clientConfig = $clientConfig;
+        $this->debugOutput = $debugOutput;
+    }
+
+    /**
+     * @param array $items
+     * @return bool
+     */
+    public function send(array $items)
+    {
+        $start = microtime(true);
+        try {
+            $this->getTopic()->publishBatch($items);
+        } catch (\Exception $e) {
+            fwrite(STDERR, $e->getMessage() . PHP_EOL);
+            return false;
+        }
+        $end = microtime(true);
+        if ($this->debugOutput) {
+            printf(
```

Comment:
Re: these printf calls

It might be better to write to STDERR because if you don't run the daemon, those output will be on the end user's web browser.

Yeah, the psrBatchLogger has the same issue. I would fix it there too.